internal/elasticsearch/test/httptest.go

```diff
@@ -61,7 +62,11 @@ func recordRequest(t *testing.T, r *http.Request, path string) {
 	require.NoError(t, err)
 
 	t.Logf("Recording %s in %s", r.URL.Path, path)
-	req, err := http.NewRequest(r.Method, r.URL.Path, nil)
+	var recordURL url.URL
+	recordURL.Path = r.URL.Path
+	recordURL.RawQuery = r.URL.RawQuery
+
+	req, err := http.NewRequest(r.Method, recordURL.String(), nil)
```

Comment:
Fix to correctly record responses of urls with query parameters. I can move this small change to a different PR if needed.

Reply:
Yes, IIRC the idea is to remove anything that is not needed from the original request. Hosts, authentication and so on are provided by the elasticsearch client.
This has limitations, as we have found here. If we had things based on headers it would fail too. Also if we would like to capture requests with payload.
We may continue improving it as we need it.